### PR TITLE
Feature/api code generation as gradletask

### LINKED ISSUE #86: Add API code generation as gradle task
**Describe the problem or limitation you are having in your project:**

For the merge purpose, we decided to set api code generation as a gradle task.

**Describe how this feature / enhancement will help you overcome this problem or limitation:**

Not appropriate

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**

See @raniejade repository

**Describe implementation detail for your proposal (in code), if possible:**

Move code to `buildSrc` of project.
Add a task implementation.
Call task from `godot-library` gradle.

**If this enhancement will not be used often, can it be worked around with a few lines of code?:**

Not appropriate.

**Is there a reason why this should be in this project and not individually solved?:**

Not appropriate.

### PR DESCRIPTION
**Pull request by piiertho**
This resolves #86 
It add api code generation, code will not compile as it is missing core types.
Also the api code generation would be changed during merge.
It setups source sets for `godot-library`.

### CODE REVIEW

File: buildSrc/src/main/kotlin/godot/codegen/Argument.kt
Comment (raniejade) [inline buildSrc/src/main/kotlin/godot/codegen/Argument.kt:7]: indentation feels a bit odd, should this be 4 spaces instead of 8?

File: buildSrc/src/main/kotlin/godot/codegen/Class.kt
Comment (raniejade) [inline buildSrc/src/main/kotlin/godot/codegen/Class.kt:10]: same here (odd indentation)

**Review by raniejade — CHANGES_REQUESTED** [review]
Looking, got a few comments and I think a few files in `.idea` is accidentally committed? `.idea/artifacts/*` for example.

File: buildSrc/build.gradle.kts
Comment (chippmann) [inline buildSrc/build.gradle.kts:13]: This version is extremely old (you see the commented 5.2. I did this when we where taking over the project. This was the current version back then). I would at least update it to the current version. But personally i would rather use something more well known like Gson, Moshi or jackson

File: buildSrc/build.gradle.kts
Comment (chippmann) [inline buildSrc/build.gradle.kts:16]: I would add the const variable in that file as well and add the comment there as well (even though it's not used at the moment. Just so the person that adds that const variable could forget that this exists or simply don't know.

File: buildSrc/src/main/kotlin/godot/codegen/Class.kt
Comment (chippmann) [reply buildSrc/src/main/kotlin/godot/codegen/Class.kt:10]: In this case the indentation comes from the fact that the closing bracket of the constructor is not on a new line. That was probably also the case on the one above but the file wasn't auto formatted afterwards.
@piiertho I suggest that you activate the checkbox "format code" inside the commit dialog in Intellij  so you don't have to remember to format each file :-)

File: godot-kotlin/godot-library/build.gradle.kts
Comment (chippmann) [inline godot-kotlin/godot-library/build.gradle.kts:11]: Targets are commented

File: godot-kotlin/godot-library/src/main/nativeInternal/kotlin/godot/internal/Utils.kt
Comment (chippmann) [inline godot-kotlin/godot-library/src/main/nativeInternal/kotlin/godot/internal/Utils.kt:14]: Please call it `getMethodBind` ^^ At first glance i always think "What? what MB? how many MB's? get 50 MB's of ram or what?" xD

Comment (chippmann) [reply godot-kotlin/godot-library/src/main/nativeInternal/kotlin/godot/internal/Utils.kt:14]: Also pleas write out the parameter names

File: godot-kotlin/godot-library/src/main/nativeInternal/kotlin/godot/internal/Utils.kt
Comment (chippmann) [inline godot-kotlin/godot-library/src/main/nativeInternal/kotlin/godot/internal/Utils.kt:19]: please write out the parameter names

File: godot-kotlin/godot-library/src/main/nativeInterop/cinterop/registrations.h
Comment (chippmann) [inline godot-kotlin/godot-library/src/main/nativeInterop/cinterop/registrations.h:1]: why do we still need this file?

File: godot-kotlin/godot-library/src/main/nativeInterop/cinterop/godot.def
Comment (chippmann) [inline godot-kotlin/godot-library/src/main/nativeInterop/cinterop/godot.def:2]: same question: why do we still need the registrations.h file?

**Review by chippmann — CHANGES_REQUESTED** [review]
I didn't look at the actual codeGen logic yet (I would still wait until it compiles with the actual merge).
But one question about it: am i blind or didn't you yet add the changes regarding inline classes that we talked about on discord?

File: godot-kotlin/godot-library/build.gradle.kts
Comment (piiertho) [reply godot-kotlin/godot-library/build.gradle.kts:11]: oops, missed this

**Comment by piiertho** [comment]
I close this one, it will be seperated into two PR, one for source sets #88 and one for api code generation.

**Pull request closed without merging**
